homeassistant/components/yolink/binary_sensor.py

```diff
@@ -153,4 +145,17 @@ def update_entity_state(self, state: dict[str, Any]) -> None:
         self._attr_is_on = self.entity_description.value(
             state.get(self.entity_description.state_key)
         )
+        # check door sensor paired device
+        if (
+            self.coordinator.device.parent_id is not None
+            and self.coordinator.device.parent_id != "null"
```

Comment:
`"null"` should be converted to `None` by the library data model. `pydantic` can handle this.

Reply:
`device` is not a data mode class, this class contains several functions for call device, should i use `pydantic` for non-data mode class?

Reply:
Why is the parent id set to `"null"`? Where does that come from?

Reply:
it's come from api response，i think  i should add device mode class for checking api response before device property was set.

Reply:
Yeah, the benefit of using pydantic is that we can validate and convert all api responses to useful objects that are typed. It's good to try to do this as early as possible when receiving the response to have as much benefit of it as possible.